utils/utils.go

```diff
@@ -461,6 +466,38 @@ func GetOpenPort(addr string, port uint32, state *State, sshConn *SSHConnection)
 			bindAddr = ""
 		}
 
+		getPortFromAuthSettings := func() uint32 {
+			holderLock.Lock()
+			defer holderLock.Unlock()
+			for _, i := range certHolder {
+				if bytes.Equal([]byte(sshConn.SSHConn.Permissions.Extensions["pubKey"]), i.key.Marshal()) {
+					for _, o := range i.options {
+						if !strings.HasPrefix(o, "sishport") {
```

Comment:
In order to stay compliant with the `sshd(8)` specification for the authorized_keys file, why don't we use the `permitlisten` option? Documentation [here](https://man.openbsd.org/OpenBSD-current/man8/sshd.8#permitlisten=__host:_port_).